lightningd/peer_control.c

```diff
@@ -848,21 +848,20 @@ static enum watch_result funding_depth_cb(struct lightningd *ld,
 					   unsigned int depth)
 {
 	const char *txidstr;
+	struct short_channel_id scid;
 
 	txidstr = type_to_string(tmpctx, struct bitcoin_txid, txid);
 	log_debug(channel->log, "Funding tx %s depth %u of %u",
 		  txidstr, depth, channel->minimum_depth);
 	tal_free(txidstr);
 
-	bool local_locked = depth >= channel->minimum_depth;
-
-	/* If we restart, we could already have peer->scid from database */
-	if (local_locked && !channel->scid) {
+	/* Reorg can change scid, so update scid when possible (depth=0 means
+	 * the stale block with our funding tx was removed) */
+	if (depth >= channel->minimum_depth || (channel->scid && depth)) {
```

Comment:
Ok, but to be honest I find the term `local_locked` a little misleading here (in `peer_control`), because it can be confused with `peer->funding_locked[LOCAL]` in `channeld`.

How about:
rename `local_locked` to `min_depth_reached`
and
`if ((min_depth_reached && !channel->scid) || (depth && channel->scid))`

Reply:
> Comparing with restarting Channeld, what do you think of failing the channel directly when we monitor the reorg?

One of the benefits of this PR is we don't need to fail the channel if both sides see the reorg. A public channel will fail anyway when either side doesn't see the reorg when reaching announce_depth (because then `scid`s don't match).

Reply:
Thank you, now I understand :)
And your name is good!